### PR TITLE
Bono/goth 378

### PR DESCRIPTION
**Pull request by bonomite**
- newsletter & form front end and alt dark version
- hook up newsletter to backend
- NEWSLETTER_API added to nuxt config and circleci files
- created re-usable EmailCollectorForm.vue comp
- updated DS to support dark mode for the checkbox, inputTextField and button-link
- hack for v-card image issue where the image does not load on desktop break point (will investigate at some point)
- more unique footer ID and Class names
- footer responsive breakpoint adjustments

### CODE REVIEW

File: assets/scss/global.scss [hunk 1]
```diff
@@ -143,3 +143,11 @@ ol ol li {
     text-transform: uppercase;
   }
 }
+
+// to go is DS extentions
+
+// v-card image hack to figure out
```
Comment (bonomite) [inline assets/scss/global.scss:149]: v-card hack to solve image issue. Will investigate at some point

File: nuxt.config.ts [hunk 2]
```diff
@@ -64,6 +64,8 @@ export default defineNuxtConfig({
     API_URL: process.env['API_URL'] || 'https://cms.demo.nypr.digital/api/v2',
     IMAGE_BASE_URL: process.env['IMAGE_BASE_URL'] || 'https://cms.demo.nypr.digital/images/',
     GA_MEASUREMENT_ID: process.env['GA_MEASUREMENT_ID'],
+    NEWSLETTER_API: process.env['NEWSLETTER_API'] || 'https://api.demo.nypr.digital/email-proxy/subscribe',
```
Comment (bonomite) [inline nuxt.config.ts:67]: Submitting the form only works with the PROD API

File: pages/index.vue [hunk 3]
```diff
@@ -12,6 +11,15 @@ const articles = await findArticlePages('').then(({ data }) =>
   normalizeFindArticlePagesResponse(data)
 )
 
+const newsletterSubmitEvent = (e) => {
+  //emitted newsletter submit event, @Matt, not exactly sure how to get this work like you mentioned.
```
Comment (bonomite) [inline pages/index.vue:15]: @walsh9 Was having trouble with the gtag event here... but I emitted it up to the page like you asked.

File: plugins/analytics.client.ts [hunk 4]
```diff
@@ -1,28 +1,27 @@
 import { useMembershipStatus } from "~~/composables/states"
 
 export default defineNuxtPlugin(() => {
-    const config = useRuntimeConfig()
-    const membershipStatus = useMembershipStatus()
+  const config = useRuntimeConfig()
```
Comment (bonomite) [inline plugins/analytics.client.ts:4]: formatting noise... we need to get on the same page with this soon.

File: components/NewsletterHome.vue [hunk 5]
```diff
@@ -0,0 +1,107 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useRuntimeConfig } from '#app'
+import axios from 'axios'
```
Comment (walsh9) [inline components/NewsletterHome.vue:4]: It seems like overkill to include all of axios just for the newsletter, I think we cant probably hand the form submission with a regular fetch.

File: components/NewsletterHome.vue [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+const hideComp = () => {
+  showComp.value = false
+}
+// submit the newsletter form and add email address to the Gothamist Newsletter list
+const submitForm = (email) => {
+  isSubmiting.value = true
+  submissionStatus.value = null
+  axios
+    .post(config.NEWSLETTER_API, {
+      list: '65dbec786b',
```
Comment (walsh9) [inline components/NewsletterHome.vue:47]: This list id shouldn't be hardcoded here, we should put it in a runtime config variable. (If it doesn't change between demo and prod, we don't need to make a whole env var for it, just moving it to the config is fine)

File: components/EmailCollectorForm.vue [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        <small
+          v-if="props.submissionStatus === 'error'"
+          id="email-address-field"
+          class="p-error mt-1 block"
+          >Sorry, there was an error with your submission. Please try
+          again!</small
+        >
+        <div class="field-checkbox mt-3">
+          <Checkbox
+            id="binary"
```
Comment (walsh9) [inline components/EmailCollectorForm.vue:110]: We should give this checkbox a better name than binary. what is it for?

File: components/GothamistFooter.vue [hunk 8]
```diff
@@ -17,13 +16,17 @@ const copyrightYear = ref(props.navData.copyrightYear)
 </script>
 
 <template>
-  <section id="footer" class="footer" data-style-mode="dark">
+  <section
+    id="gotahmist-footer"
```
Comment (walsh9) [inline components/GothamistFooter.vue:20]: Looks like a typo here.

File: components/ScrollToTopButton.vue [hunk 9]
```diff
@@ -22,7 +22,7 @@ const initScrollTrigger = () => {
   if (!props.hide) {
     gsapScrollTrigger = ScrollTrigger.create({
       id: 'scrollTopID',
-      trigger: '#footer',
+      trigger: '#gotahmist-footer',
```
Comment (walsh9) [inline components/ScrollToTopButton.vue:25]: A matching typo for the earlier one.

File: components/NewsletterHome.vue [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    //default: 'pi-arrow-right',
+    default: null,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'submitEventEmit', value: any): void
+}>()
+
+const showComp = ref(true)
```
Comment (walsh9) [inline components/NewsletterHome.vue:33]: Can we spell out the word "Component" here. It's not immediately obvious if this is referring to component or composition or completion or something else here.

File: components/NewsletterHome.vue [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+const submissionStatus = ref(null)
+const isSubmiting = ref(false)
```
Comment (walsh9) [inline components/NewsletterHome.vue:35]: another typo here, should be submitting with two t's

**Review by walsh9 — CHANGES_REQUESTED** [review]
Looks good overall but I did leave a lot of feedback about names. It might seem like nitpicking but we're gonna have to read this stuff later so I think it's important.

File: components/EmailCollectorForm.vue [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+  thanksMessage: {
+    type: String,
+    default: 'Thank you for signing up!',
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'submitFormEmit', value: any): void
+  (e: 'noThanksEventEmit'): void
+}>()
```
[the remark this answers is not in the document]
Comment (bonomite) [reply components/EmailCollectorForm.vue:38]: ok... I understand. Cool. I guess there is no need to mention it is an emit... because it's clearly an emit... and as a prop, it's preceded with a "@" which clear indicates its is an emit.

File: components/NewsletterHome.vue (hunk 6 above)
Comment (bonomite) [reply components/NewsletterHome.vue:47]: done

File: components/GothamistFooter.vue (hunk 8 above)
Comment (bonomite) [reply components/GothamistFooter.vue:20]: whoops!

File: components/EmailCollectorForm.vue (hunk 7 above)
Comment (bonomite) [reply components/EmailCollectorForm.vue:110]: not needed, I removed the id prop

**Comment by bonomite** [comment]
Great stuff @walsh9 !! Appreciate looking through. I made updates and covered all your comments except the Axios replacement. 
I also realized that I did not add a email format validation method... so I added that to the EmailCollectorForm.vue file. and also changed the logic on how the error text populates. I'll see if I can get the Fetch() in there Monday morning. Have a great long weekend.

File: components/NewsletterHome.vue [hunk 13]
```diff
@@ -41,17 +40,20 @@ const hideComp = () => {
...
+    body: JSON.stringify({
       list: config.NEWSLETTER_LIST_ID,
       email: email,
-      headers: {
-        'Content-Type': 'application/json',
-        'Access-Control-Allow-Origin': '*',
-      },
-    })
+    }),
+  })
```
Comment (walsh9) [inline components/NewsletterHome.vue:55]: I think if you use nuxt's $fetch you can shorten this to just something like
```  $fetch(config.NEWSLETTER_API, {
      method: 'POST',
      body: {
        list: config.NEWSLETTER_LIST_ID,
        email: email,
      }
  })
```
https://v3.nuxtjs.org/api/utils/$fetch
https://github.com/unjs/ohmyfetch#%EF%B8%8F-json-body

**Review by walsh9 — CHANGES_REQUESTED** [review]
Looks great. Just one last thing

File: components/NewsletterHome.vue (hunk 13 above)
Comment (bonomite) [reply components/NewsletterHome.vue:55]: if I remove the "JSON.stringify()" I get this error in VSCODE
image

Comment (bonomite) [reply components/NewsletterHome.vue:55]: but it seems to submit fine....

Comment (bonomite) [reply components/NewsletterHome.vue:55]: ah... ok added the "$" to the fetch

**Review by walsh9 — APPROVED** [review]
